askama_parser/src/lib.rs

```diff
@@ -1,3 +1,13 @@
+//! Parser and syntax tree for Askama's template syntax.
+//!
+//! Askama templates are a series of [`node::Node`s](./node/enum.Node.html),
+//! which represent either some literal text in the template or one of the
+//! three types of template tags: expressions, blocks, or comments.
+//!
+//! The main entry point to this crate is the [`parse()`](./fn.parse.html)
```

Comment:
I don't think this is a very nice public API. Maybe change it to wrap a `struct Ast<'a> { nodes: Vec<Node<'a>> }` and change `parse()` to become `Ast::from_str()`? (In a separate commit.)

Reply:
I don't like `Template` since that's already what the trait is called.

Reply:
Initially I was thinking `std::str::FromStr`, but upon review I noticed this:

> FromStr does not have a lifetime parameter, and so you can only parse types that do not contain a lifetime parameter themselves.

Perhaps you mean a non-trait `from_str` method?  Since it also takes a `Syntax` parameter, I'm wondering if `from_str` fully captures the meaning.

As to the new struct's name, some things stood out.  The `parse_template` helper is what produces this top-level result and the bodies of loops, if and else clauses, etc. and it makes sense to unify the concepts.  Those elements are frequently referred to as blocks.  Perhaps `struct Block<'a> { nodes : Vec<Node<'a>> }`?